src/App.jsx

```diff
@@ -1,24 +1,103 @@
-import logo from "./assets/technigo-logo.svg";
-// Import the custom hook useQRCodeGenerator
-import { QrExample } from "./components/QrExample";
+import logo from "./assets/technigo-logo-white.svg";
+import { ColorSelect } from "./components/ColorSelect";
+import { QrAnimation, LoadingAnimation } from "./components/LottieComponent";
+import { useQRCodeGenerator } from "./hooks/useQRCodeGenerator";
 
-// Define the App component
 export const App = () => {
-  // Destructure variables, properties and methods from the useQRCodeGenerator hook that you imported above here :)
+    const {
+        setUrl,
+        url,
+        qr,
+        showInput,
+        downloadQRCode,
+        generateQRCode,
+        repeatAction,
+        color,
+        setColor,
+        isLoading,
+    } = useQRCodeGenerator();
 
-  // Return the JSX to render the component
-  return (
-    <div className="">
-      {/* Render the title */}
-      <img className="logo" src={logo} alt="" />
-      <h1>Technigo QR Code Generator</h1>
-      <p>Start Here</p>
-      <QrExample />
+    const handleColorChange = (e) => {
+        const selectedColor = e.target.value;
+        console.log("Color changed:", selectedColor);
+        setColor(selectedColor);
+    };
 
-      {/* Conditionally render based on wether the user is inputting an URL to generate a QR Code or the user wnats to downaload the generated QR Code from the url input */}
-      {/* {yourReactiveVariableThatTogglesTheDownloadQrCcodeOrInputField ? () : ()} */}
-    </div>
-  );
+    return (
+        <div className="app">
+            <div className="app-container">
+                <div className="header">
+                    <img className="logo" src={logo} alt="" />
+                    <h1 className="header-h1">QR CODE GENERATOR</h1>
+                </div>
+                <div className="generator">
+                    {isLoading ? (
+                        <div className="loading-container">
+                            <LoadingAnimation />
+                        </div>
+                    ) : showInput ? (
+                        <>
+                            <QrAnimation />
+                            <h3>
+                                Want a QR code? You've come to the right place!
+                                Just paste in your url here, and click generate!
+                                You'll be able to download it once the QR code
+                                is generated.{" "}
+                            </h3>
```

Comment:
I'm not sure I understand what are the curly braces with the empty string is referring to? :) 
Also, just a tiny html tip! There is a lot of text in this H3. Is it intended to be a heading? Otherwise it's perhaps better to use a p-tag and style it in CSS :)